server_test.go

```diff
@@ -41,3 +56,145 @@ func TestParseHexColor(t *testing.T) {
 		}
 	}
 }
+
+// TestTLSAutoRegeneration creates an expired TLS certificate, to test that a
+// new TLS certificate pair is regenerated when the old pair expires. This is
+// necessary because the pair expires after a little over a year.
+func TestTLSAutoRegeneration(t *testing.T) {
+
+	homePath, err := os.UserHomeDir()
+	if err != nil {
+		t.Fatalf("couldn't retrieve TLS config")
+	}
+	certPath := homePath + "/.lnd/tls.cert"
+	keyPath := homePath + "/.lnd/tls.key"
```

Comment:
A temporary file should be used. Otherwise, this _will_ mutate the existing certs when run on a machine with an `lnd` node initialized.

Reply:
@orbitalturtle you can create a temporary directory using [`ioutil.TempDir`](https://golang.org/pkg/io/ioutil/#TempDir) and put the files in there. see the example for how to also cleanup the directory